### PR TITLE
Create search API

### PR DESCRIPTION
**Pull request by cthoyt**
This PR adds a search API for nodes in the DKG. It's implemented using Cypher's `CONTAINS` syntax applied to names and synonyms combine with a small amount of string processing (case normalization, hyphen normalization, space normalization)

### CODE REVIEW

File: mira/dkg/client.py [hunk 1]
```diff
@@ -217,6 +217,24 @@ def get_node_counter(self) -> Counter:
...
+                replace(replace(toLower(n.name), '-', ''), '_', '') CONTAINS '{query_lower}'
+                OR any(
+                    synonym IN n.synonyms 
+                    WHERE replace(replace(toLower(synonym), '-', ''), '_', '') CONTAINS '{query_lower}'
+                )
+            RETURN n 
+            LIMIT {limit}
+        """
+        )
+        return [Entity(**n) for n in self.query_nodes(cypher)]
```
Comment (bgyori) [inline mira/dkg/client.py:236]: I am wondering if we should try to do some sorting/ranking at this point - perhaps a simple string similarity score to rank results?

Comment (cthoyt) [reply mira/dkg/client.py:236]: That's possible using the built-in diff library or something external like `fuzzywuzzy`. Looking forward, there are for sure a lot of things to be done to improve the goodness of these (some task queues like Redis have support for this, elastic search, etc.)

Comment (bgyori) [reply mira/dkg/client.py:236]: Yes, I think we could use fuzzywuzzy, I think it's a simple and useful first approach to get a scored ranking.

Comment (cthoyt) [reply mira/dkg/client.py]: Done in 6db4a52 with built-in `difflib`

**Review by bgyori — APPROVED** [review]

**Pull request merged**